### PR TITLE
added paddle.sum frontend and test for it

### LINKED ISSUE #22770: sum

### PR DESCRIPTION
**Pull request by sabre-code**
Added paddle.sum frontend function. Wrote tests for it. 

Fixes #22770

### CODE REVIEW

**Comment by sabre-code** [comment]
@yopknopixx hello please review my PR.

**Comment by shubhamb-22** [comment]
@sabre-code I see a lot of new failures introduced by your PR. [https://github.com/unifyai/ivy/actions/runs/6021337957?pr=22771](url)

**Comment by sabre-code** [comment]
@shubhamb-22 I guess they are from some other functions. Can you check code.

**Comment by sabre-code** [comment]
@shubhamb-22 Please guide with possible issue. This is my first PR here.

**Comment by yopknopixx** [comment]
@sabre-code I will go through it shortly

**Comment by sabre-code** [comment]
> @sabre-code I will go through it shortly

Thanks

File: ivy/functional/frontends/paddle/tensor/math.py [hunk 1]
```diff
@@ -487,6 +487,17 @@ def subtract(x, y, name=None):
     return ivy.subtract(x, y)
 
 
+@with_supported_dtypes(
+    {"2.5.1 and below": ("bool", "int32", "int64", "float32", "float64")}, "paddle"
+)
+@to_ivy_arrays_and_back
+def sum(x, axis=None, keepdim=False, name=None):
+    if x.dtype == ivy.int32:
```
Comment (shubhamb-22) [inline ivy/functional/frontends/paddle/tensor/math.py:495]: What is the reason for doing this?

Comment (yopknopixx) [inline ivy/functional/frontends/paddle/tensor/math.py:495]: What is the reason for doing this?

Comment (sabre-code) [reply ivy/functional/frontends/paddle/tensor/math.py:495]: @yopknopixx the paddle documentation says if the dtype of input array is int32 the output dtype should be int64. ivy.sum does not cover this edge case and if this dtype conversion is not done all tests fail with "expected dtype int64 got int32". to address this issue this is done. Thanks

Comment (sabre-code) [reply ivy/functional/frontends/paddle/tensor/math.py]: @yopknopixx I rechecked and found that even when input array dtype is bool paddle.sum converts it to int64, I have added condition for that too. and ivy.sum documentation says whenever such expected dtype is different from dtype of array we have to convert the dtype to desired type before passing to the function. so added code to do that too.

**Comment by Omar88magdy** [comment]
hey sabre-code, 
i want to inform you that i've been working on (sum) function before you and i 've already implemented it.
thank you.

**Comment by sabre-code** [comment]
> hey sabre-code, i want to inform you that i've been working on (sum) function before you and i 've already implemented it. thank you.

apologies, I guess this happened because to-do list was't updated when I started working on it.
